File: Java_Design-Patterns/src/adapter/MotorElectricoAdapter.java
package adapter;

public class MotorElectricoAdapter extends Motor {

    private MotorElectrico motorElectrico;

    public MotorElectricoAdapter(){
        super();
        this.motorElectrico = new MotorElectrico();
    }

    public void encender(){
        this.motorElectrico.conectar();
        this.motorElectrico.activar();
    }

    public void acelerar(){
        this.motorElectrico.moverMasRapido();
    }

    public void apagar(){
        this.motorElectrico.detener();
        this.motorElectrico.desconectar();
    }

}

File: Python_Design-Patterns/adapter/MotorElectricoAdapter.py
from MotorElectrico import *
from Motor import *

class MotorElectricoAdapter(Motor):

    def __init__(self):
        self.motorElectrico = MotorElectrico()

    def encender(self):
        self.motorElectrico.conectar()
        self.motorElectrico.activar()

    def acelerar(self):
        self.motorElectrico.moverMasRapido()

    def apagar(self):
        self.motorElectrico.detener()
        self.motorElectrico.desconectar()